### PR TITLE
Check sendMessageBatch result, log if there are failures.

### PR DESCRIPTION
**Pull request by xiaodong-izettle**
How about something like this, @calle? Thanks...

### CODE REVIEW

**Comment by calle** [comment]
Yes, this is better than silently ignoring them. Maybe add the special case of sending only a single message and throw an exception to allow the caller to handle it better?

This is what is invoked when you invoke `post(T message, String eventName)` (and done when publishing card payments from the planet to the sun.

**Comment by calle** [comment]
Also some tests are failing

**Comment by xiaodong-izettle** [comment]
Hehe, sure, thanks, @calle. Will look into it... 👍

**Comment by xiaodong-izettle** [comment]
Ping @calle, Oskar had already added `post(T message, String eventName)` method... 

Here I just add logging for batch operations...

File: izettle-messaging/src/main/java/com/izettle/messaging/QueueServiceSender.java [hunk 1]
```diff
@@ -192,7 +201,34 @@ private String wrapInSNSMessage(
 
     private void sendMessageBatch(Collection<SendMessageBatchRequestEntry> messages) {
         for (Collection<SendMessageBatchRequestEntry> batch : partition(messages, MAX_BATCH_SIZE)) {
-            amazonSQS.sendMessageBatch(new SendMessageBatchRequest(queueUrl, new ArrayList<>(batch)));
+            final SendMessageBatchResult sendMessageBatchResult =
+                amazonSQS.sendMessageBatch(new SendMessageBatchRequest(queueUrl, new ArrayList<>(batch)));
+            final List<BatchResultErrorEntry> failed = sendMessageBatchResult.getFailed();
+            if (!failed.isEmpty()) {
```
Comment (calle) [inline izettle-messaging/src/main/java/com/izettle/messaging/QueueServiceSender.java]: Any chance of a `NullPointerException` here? Or will `getFailed` always return a valid List?

Comment (xiaodong-izettle) [inline izettle-messaging/src/main/java/com/izettle/messaging/QueueServiceSender.java]: Hehe, thanks, @calle... 

Based on the source code, it's always a `List`... :)

Comment (calle) [inline izettle-messaging/src/main/java/com/izettle/messaging/QueueServiceSender.java]: The best type of argument! 😄

**Comment by calle** [comment]
What do you say about throwing a `MessagingException` instead? A partial failure should probably be handled in the same way as a full failure right?

**Comment by xiaodong-izettle** [comment]
Yeah, I was considering that... 

Then I also consider that should we keep the behaviour the same at first, to have an impression about the failure rate first? 

I don't know, what do you say? :-)

**Comment by calle** [comment]
Let's start with logging since Oskar already updated the method planet uses (single message) to throw on errors and that was the important thing to do (to notice when these publications failed).

**Comment by xiaodong-izettle** [comment]
Hehe, thanks, @calle. :-)

**Pull request merged**